test/e2e/local-cluster/create_cluster.go

```diff
@@ -0,0 +1,229 @@
+	flag.StringVar(&kubeconfigInternalName, "kubeconfigInternalName", "e2e.internal.kubeconfig", "kubeconfig file name for connecting to the e2e clusters from the same Docker network")
+
+	flag.StringVar(&dockerImageRepository, "dockerImageRepository", "greenhouse", "Docker image repository  for Greenhouse manager")
+	flag.StringVar(&dockerImageTag, "dockerImageTag", "e2e-latest", "Docker image tag for Greenhouse manager")
+	flag.BoolVar(&dockerImageBuildSkip, "dockerImageBuildSkip", false, "Skip building the docker image for Greenhouse manager")
+
+	flag.StringVar(&greenhouseControllerManagerNamespace, "greenhouseControllerManagerNamespace", "greenhouse", "Namespace for deploying Greenhouse manager")
+	flag.StringVar(&greenhouseControllerManagerRelease, "greenhouseControllerManagerRelease", "greenhouse", "Helm release name for deploying Greenhouse manager")
+
+	flag.Parse()
+
+	l.Info("configuration loaded", "kindClusterName", kindClusterName, "dockerImageRepository", dockerImageRepository, "dockerImageTag", dockerImageTag, "dockerImageBuildSkip", dockerImageBuildSkip, "kubeconfigName", kubeconfigName, "kubeconfigInternalName", kubeconfigInternalName)
+
+}
+
+func main() {
+
+	ctx := context.Background()
+	l := log.FromContext(ctx)
+
+	// Create cluster
+	cluster := kind.NewCluster(kindClusterName)
+	cluster.SetDefaults()
+	kubeconfig, err := cluster.Create(ctx)
+	if err != nil {
+		l.Error(err, "cluster creation")
+		os.Exit(1)
+	}
+	l.Info("cluster created successfully")
+
+	// Export kubeconfig
+	f, err := os.Create(kubeconfigName)
+	if err != nil {
+		l.Error(err, "kubeconfig creation")
+		os.Exit(1)
+	}
+	args := []string{"export", "kubeconfig", "--name", kindClusterName, "--kubeconfig", f.Name()}
+	cmd := exec.Command("kind", args...)
+	_, err = cmd.Output()
+	if err != nil {
+		l.Error(err, "kubeconfig export")
+		os.Exit(1)
+	}
+	f, err = os.Create(kubeconfigInternalName)
+	if err != nil {
+		l.Error(err, "kubeconfig creation")
+		os.Exit(1)
+	}
+	args = []string{"export", "kubeconfig", "--name", kindClusterName, "--internal", "--kubeconfig", f.Name()}
+	cmd = exec.Command("kind", args...)
+	_, err = cmd.Output()
+	if err != nil {
+		l.Error(err, "kubeconfig export")
+		os.Exit(1)
+	}
+
+	l.Info("kubeconfig exported successfully")
+
+	// Build image
+	image := fmt.Sprintf("%s:%s", dockerImageRepository, dockerImageTag)
+	if !dockerImageBuildSkip {
+		err = dockerImageBuild("./../../../", image)
+		if err != nil {
+			l.Error(err, "image build")
+			os.Exit(1)
+		}
+		l.Info("Docker image built successfully")
+	}
+
+	// Load image
+	err = cluster.LoadImage(ctx, image)
+	if err != nil {
+		l.Error(err, "image load")
+		os.Exit(1)
+	}
+	l.Info("Docker image loaded to the cluster successfully")
+
+	// Deploy Greenhouse manager
+	err = installChart(ctx, "./../../../charts/manager", greenhouseControllerManagerRelease, kubeconfig, greenhouseControllerManagerNamespace)
```

Comment:
I think we also need to install the [DEX (idproxy) CRDs](https://github.com/cloudoperators/greenhouse/tree/main/charts/idproxy/crds), the controller won't come up for me:
```
....
2024-07-03T08:43:10Z	ERROR	controller-runtime.source.EventHandler	if kind is a CRD, it should be installed before calling Start	{"kind": "Connector.dex.coreos.com", "error": "no matches for kind \"Connector\" in version \"dex.coreos.com/v1\""}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/internal/source/kind.go:63
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
	/go/pkg/mod/k8s.io/apimachinery@v0.29.5/pkg/util/wait/loop.go:87
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
	/go/pkg/mod/k8s.io/apimachinery@v0.29.5/pkg/util/wait/loop.go:88
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
	/go/pkg/mod/k8s.io/apimachinery@v0.29.5/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/internal/source/kind.go:56
2024-07-03T08:43:10Z	ERROR	Could not wait for Cache to sync	{"controller": "organizationDEX", "controllerGroup": "greenhouse.sap", "controllerKind": "Organization", "error": "failed to wait for organizationDEX caches to sync: timed out waiting for cache to be synced for Kind *api.Connector"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/internal/controller/controller.go:203
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/internal/controller/controller.go:208
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/internal/controller/controller.go:234
sigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.5/pkg/manager/runnable_group.go:223
2024-07-03T08:43:10Z	INFO	Stopping and waiting for non leader election runnables
2024-07-03T08:43:10Z	INFO	Stopping and waiting for leader election runnables
2024-07-03T08:43:10Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "clusterStatus", "controllerGroup": "greenhouse.sap", "controllerKind": "Cluster"}
2024-07-03T08:43:10Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "pluginHelm", "controllerGroup": "greenhouse.sap", "controllerKind": "Plugin"}
2024-07-03T08:43:10Z	INFO	Shutdown signal received, waiting for all workers to finish	{"controller": "pluginPreset", "controllerGroup": "greenhouse.sap", "controllerKind": "PluginPreset"}
....
```

Reply:
@uwe-mayer  Should we deploy only CRDs or idproxy Helm chart (with scaled to zero)?

Reply:
done with cd9b6279c0847fdd58c0d4105dd58d539a8d293c 

manager pod starts as expected:

```
kubectl logs greenhouse-controller-manager-bb9756c78-947xx
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "teamRoleBindingController"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "pluginPreset"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "bootStrap"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "organizationTeamRoleSeeder"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "teamMembershipPropagation"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "pluginHelm"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "clusterDirectAccess"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "clusterStatus"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "organizationDEX"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "teamPropagation"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "clusterHeadscaleAccess"}
2024-07-11T11:32:19Z    INFO    setup   WARN: headscaleApiKey or headscaleApiUrl is not set
2024-07-11T11:32:19Z    INFO    setup   Skipping headscale access reconciler
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "organizationNamespace"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "organizationRBAC"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "organizationServiceProxy"}
2024-07-11T11:32:19Z    INFO    setup   registering controller  {"name": "teamCAM"}
2024-07-11T11:32:19Z    INFO    setup   registering webhook     {"name": "cluster"}
2024-07-11T11:32:19Z    INFO    controller-runtime.builder      Registering a mutating webhook    {"GVK": "greenhouse.sap/v1alpha1, Kind=Cluster", "path": "/mutate-greenhouse-sap-v1alpha1-cluster"}
2024-07-11T11:32:19Z    INFO    controller-runtime.webhook      Registering webhook       {"path": "/mutate-greenhouse-sap-v1alpha1-cluster"}
2024-07-11T11:32:19Z    INFO    controller-runtime.builder      Registering a validating webhook  {"GVK": "greenhouse.sap/v1alpha1, Kind=Cluster", "path": "/validate-greenhouse-sap-v1alpha1-cluster"}
2024-07-11T11:32:19Z    INFO    controller-runtime.webhook      Registering webhook       {"path": "/validate-greenhouse-sap-v1alpha1-cluster"}
2024-07-11T11:32:19Z    INFO    setup   registering webhook     {"name": "pluginDefinition"}
2024-07-11T11:32:19Z    INFO    controller-runtime.builder      Registering a mutating webhook    {"GVK": "greenhouse.sap/v1alpha1, Kind=PluginDefinition", "path": "/mutate-greenhouse-sap-v1alpha1-plugindefinition"}
2024-07-11T11:32:19Z    INFO    controller-runtime.webhook      Registering webhook       {"path": "/mutate-greenhouse-sap-v1alpha1-plugindefinition"}
...
```